core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -111,7 +111,7 @@ public QueryValidationException validate() {
     }
 
     @Override
-    public String getName() {
+    public String getWriteableName() {
```

Comment:
I got sidetracked while wondering if we should make all these methods `final` so no subclass can change the name. Now I'm unsure how subclassing queries would work in terms of name, writeTo, but especially readFrom(). Suppose we have a `SpecialAndQueryBuilder extends AndQueryBuilder`, having its own WritableName is okay, also `doWriteTo` could call `super.doWriteTo()`, but as far as I can see we'd have to repeat the `readFrom` code in any subclass since the superclass always return their own object type. Or is this something we just assume we won't have to do for other NamedWritables?